controllers/vspheremachine_controller.go

```diff
@@ -316,6 +362,13 @@ func (r *machineReconciler) reconcileDelete(ctx context.Context, machineCtx capv
 			return reconcile.Result{}, nil
 		}
 		conditions.MarkFalse(machineCtx.GetVSphereMachine(), infrav1.VMProvisionedCondition, clusterv1.DeletionFailedReason, clusterv1.ConditionSeverityWarning, "")
+		v1beta2conditions.Set(machineCtx.GetVSphereMachine(), metav1.Condition{
+			Type:    infrav1.VSphereMachineVirtualMachineProvisionedV1Beta2Condition,
+			Status:  metav1.ConditionFalse,
+			Reason:  infrav1.VSphereMachineVirtualMachineDeletingV1Beta2Reason,
+			Message: err.Error(),
```

Comment:
If this err is not deterministic this will lead to continuous reconciles across the entire hierarchy. I think I would add the "take a look at logs" message we usually used

Reply:
This is a pattern used in many place in CAPV, might be we should tackle this as a follow up

Reply:
The only reason I'm mentioning it is because here we are additionally adding the error compared to before. So I'm afraid of introducing a problem (and a huge one) (while everything else is not a change to before).

But up to you if you want to take the risk